src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd

```diff
@@ -624,68 +664,7 @@ The message root is 'NITF'.
-                                  </xs:element>
-                                </xs:choice>
-                              </xs:sequence>
-                            </xs:complexType>
-                          </xs:element>
-                          <xs:element name="BlockMaskRecord" minOccurs="0" maxOccurs="unbounded" type="nitf:unsignedBinaryIntBytes" dfdl:length="4" dfdl:occursCountKind="expression" dfdl:occursCount="{ if (../BlockMaskRecordLength eq 0) then 0 else (../../../Header/NumberOfBlocksPerRow * ../../../Header/NumberOfBlocksPerColumn * (if (../../../Header/ImageMode eq 'S') then (if (fn:exists(../../../Header/NumberOfMultispectralBands)) then ../../../Header/NumberOfMultispectralBands[1]/Value else ../../../Header/NumberOfBands) else xs:unsignedInt(1))) }" />
-                          <xs:element name="PadPixelMaskRecord" minOccurs="0" maxOccurs="unbounded" type="nitf:unsignedBinaryIntBytes" dfdl:length="4" dfdl:occursCountKind="expression" dfdl:occursCount="{ if (../PadOutputPixelCodeLength eq 0) then 0 else (../../../Header/NumberOfBlocksPerRow * ../../../Header/NumberOfBlocksPerColumn * (if (../../../Header/ImageMode eq 'S') then (if (fn:exists(../../../Header/NumberOfMultispectralBands)) then ../../../Header/NumberOfMultispectralBands[1]/Value else ../../../Header/NumberOfBands) else xs:unsignedInt(1))) }" />
-                        </xs:sequence>
-                      </xs:complexType>
-                    </xs:element>
-                    <xs:element name="PayloadLength" type="xs:long" dfdl:inputValueCalc="{ (../../../Header/ImageSegmentLengths[dfdl:occursIndex()]/DataLength) - (if (fn:exists(../MaskTable)) then dfdl:valueLength(../MaskTable[1], 'bytes') else 0) }" />
-                    <xs:choice dfdl:choiceDispatchKey="{ ../Header/ImageCompression }">
-                      <xs:element ref="JFIF" dfdl:choiceBranchKey="C3 M3 C5 M5 I1" dfdl:lengthKind="explicit" dfdl:length="{ ../PayloadLength }" />
-                      <xs:element name="JPEG2000" dfdl:choiceBranchKey="C8 M8" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="{ ../PayloadLength }" />
-                      <xs:element name="BiLevel" dfdl:choiceBranchKey="C1 M1" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="{ ../PayloadLength }" />
-                      <xs:element name="VectorQuantization" dfdl:choiceBranchKey="C4 M4" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="{ ../PayloadLength }" />
-                      <xs:element name="NoCompression" dfdl:choiceBranchKey="NC NM" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="{ ../PayloadLength }" />
-                    </xs:choice>
-                  </xs:sequence>
-                </xs:complexType> 
-              </xs:element>
+              <xs:element name="Data" type="ude:image_data_type" dfdl:lengthKind="explicit" dfdl:length="{ ../../Header/ImageSegmentLengths[dfdl:occursIndex()]/DataLength }" />
```

Comment:
The rest of this file uses camel case for types. For consistency, should we rename this ImageDataType?